packages/shared-preferences/android/src/main/java/io/flutter/plugins/shared_preferences/SharedPreferencesPlugin.java

```diff
@@ -20,6 +20,9 @@
   private static final String SHARED_PREFERENCES_NAME = "FlutterSharedPreferences";
   private static final String CHANNEL_NAME = "plugins.flutter.io/shared_preferences";
 
+  private static final String LIST_IDENTIFIER = "<<<LIST>>>";
```

Comment:
That's what we were doing before. It bit us because Platform Messages does not support Set. So we had to convert set back to list prior to sending it back to Flutter. Now let's see:

- We are exposing an API called setStringSet(Set) in the plugin.
- Internally, we are converting it to List to send it over to native.
   - iOS can store it as is but when it is sent back as List, we convert it back to Set on Dart side.
   - Android has to convert the List back to a Set to store it and then back to List to send it back (which then gets converted to Set on Dart side).

So I figured we can just get rid of this craziness and just support ordered storage.

- setStringList(List)
- No conversion while sending it over or when it comes back.
   - iOS stores as is.
   - Android stores an encoding.